npm/test/getfilecheckprogress.test.ts

```diff
@@ -59,56 +56,71 @@ class GraphqlClientFailure {
   const clientMock = GraphqlClient as jest.Mock
   clientMock.mockImplementation(() => new GraphqlClientSuccess())
   const client = new GraphqlClient("https://test.im", mockKeycloakInstance)
   document.body.innerHTML = `<input id="consignmentId" type="hidden" value="${consignmentId}">`
 
-  const callback: (fileChecksProgress: IFileCheckProgress | null) => boolean =
-    (fileChecksProgress) => {
-      expect(fileChecksProgress!.antivirusProcessed).toBe(2)
-      expect(fileChecksProgress!.checksumProcessed).toBe(3)
-      expect(fileChecksProgress!.ffidProcessed).toBe(4)
-      expect(fileChecksProgress!.totalFiles).toBe(10)
-      done()
-      return true
-    }
-  getFileChecksInfo(client, callback)
+  const fileChecksProgress: IFileCheckProgress | null =
+    await getFileChecksProgress(client)
+
+  expect(fileChecksProgress!.antivirusProcessed).toBe(2)
+  expect(fileChecksProgress!.checksumProcessed).toBe(3)
+  expect(fileChecksProgress!.ffidProcessed).toBe(4)
+  expect(fileChecksProgress!.totalFiles).toBe(10)
 })
 
-test("getFileChecksInfo returns null with a failed api call", (done) => {
+test("getFileChecksProgress returns null with a failed api call", async () => {
   const consignmentId = "7d4ae1dd-caeb-496d-b503-ab0e8d82a12c"
   const clientMock = GraphqlClient as jest.Mock
   clientMock.mockImplementation(() => new GraphqlClientFailure())
   const client = new GraphqlClient("https://test.im", mockKeycloakInstance)
   document.body.innerHTML = `<input id="consignmentId" type="hidden" value="${consignmentId}">`
 
-  const callback: (fileChecksProgress: IFileCheckProgress | null) => boolean =
-    (fileChecksProgress) => {
-      expect(fileChecksProgress).toBeNull()
-      done()
-      return false
-    }
+  const fileChecksProgress: IFileCheckProgress | null =
+    await getFileChecksProgress(client)
 
-  getFileChecksInfo(client, callback)
+  expect(fileChecksProgress).toBeNull()
 })
 
-test("getFileChecksInfo returns null with empty data from a successful api call", (done) => {
+test("getFileChecksProgress returns null with empty data from a successful api call", async () => {
   const consignmentId = "7d4ae1dd-caeb-496d-b503-ab0e8d82a12c"
   const clientMock = GraphqlClient as jest.Mock
   clientMock.mockImplementation(() => new GraphqlClientEmptyData())
   const client = new GraphqlClient("https://test.im", mockKeycloakInstance)
   document.body.innerHTML = `<input id="consignmentId" type="hidden" value="${consignmentId}">`
 
-  const callback: (fileChecksProgress: IFileCheckProgress | null) => boolean =
-    (fileChecksProgress) => {
-      expect(fileChecksProgress).toBeNull()
-      done()
-      return false
-    }
-  getFileChecksInfo(client, callback)
+  const fileChecksProgress: IFileCheckProgress | null =
+    await getFileChecksProgress(client)
+
+  expect(fileChecksProgress).toBeNull()
+})
+
+test("getFileChecksProgress returns null with errors from a successful api call", async () => {
```

Comment:
What case does this test cover - if the API call was a success, why should it return null? And what does "null with errors" mean?

Reply:
Because I realised that we don't have a test [for this](https://github.com/nationalarchives/tdr-transfer-frontend/blob/c83056108f273cc982e6d82df7b47e4f1c94d606/npm/src/filechecks/get-file-check-progress.ts#L37), which I thought was a large omission since it would test the case in which the GraphQL call was successful but it returned errors such as "the user does not have access to this consignment". The only tests we had were

1. GraphQL response success, response has data - the code in first `else`
2. GraphQL response failure - the code in `catch`
3. GraphQL response success, response has no data - the code in nested `else`; the one that you asked about
but no "GraphQL response success, response has errors" - the code in `if` statement

> And what does "null with errors" mean

I just copied the same format as the previous test (maybe we can reword it?). This basically means the function returns null after the successful graphql response had errors in it